src/pudl/extract/ferc1.py

```diff
@@ -64,10 +64,156 @@
+    'f1_note_fin_stmnt': 'F1_43.DBF',
+    'f1_nuclear_fuel': 'F1_44.DBF',
+    'f1_officers_co': 'F1_45.DBF',
+    'f1_othr_dfrrd_cr': 'F1_46.DBF',
+    'f1_othr_pd_in_cptl': 'F1_47.DBF',
+    'f1_othr_reg_assets': 'F1_48.DBF',
+    'f1_othr_reg_liab': 'F1_49.DBF',
+    'f1_overhead': 'F1_50.DBF',
+    'f1_pccidica': 'F1_51.DBF',
+    'f1_plant_in_srvce': 'F1_52.DBF',
+    'f1_pumped_storage': 'F1_53.DBF',
+    'f1_purchased_pwr': 'F1_54.DBF',
+    'f1_reconrpt_netinc': 'F1_55.DBF',
+    'f1_reg_comm_expn': 'F1_56.DBF',
+    'f1_respdnt_control': 'F1_57.DBF',
+    'f1_retained_erng': 'F1_58.DBF',
+    'f1_r_d_demo_actvty': 'F1_59.DBF',
+    'f1_sales_by_sched': 'F1_60.DBF',
+    'f1_sale_for_resale': 'F1_61.DBF',
+    'f1_sbsdry_totals': 'F1_62.DBF',
+    'f1_schedules_list': 'F1_63.DBF',
+    'f1_security_holder': 'F1_64.DBF',
+    'f1_slry_wg_dstrbtn': 'F1_65.DBF',
+    'f1_substations': 'F1_66.DBF',
+    'f1_taxacc_ppchrgyr': 'F1_67.DBF',
+    'f1_unrcvrd_cost': 'F1_68.DBF',
+    'f1_utltyplnt_smmry': 'F1_69.DBF',
+    'f1_work': 'F1_70.DBF',
+    'f1_xmssn_adds': 'F1_71.DBF',
+    'f1_xmssn_elc_bothr': 'F1_72.DBF',
+    'f1_xmssn_elc_fothr': 'F1_73.DBF',
+    'f1_xmssn_line': 'F1_74.DBF',
+    'f1_xtraordnry_loss': 'F1_75.DBF',
+    'f1_codes_val': 'F1_76.DBF',
+    'f1_sched_lit_tbl': 'F1_77.DBF',
+    'f1_audit_log': 'F1_78.DBF',
+    'f1_col_lit_tbl': 'F1_79.DBF',
+    'f1_load_file_names': 'F1_80.DBF',
+    'f1_privilege': 'F1_81.DBF',
+    'f1_sys_error_log': 'F1_82.DBF',
+    'f1_unique_num_val': 'F1_83.DBF',
+    'f1_row_lit_tbl': 'F1_84.DBF',
+    'f1_footnote_data': 'F1_85.DBF',
+    'f1_hydro': 'F1_86.DBF',
+    'f1_footnote_tbl': 'F1_87.DBF',
+    'f1_ident_attsttn': 'F1_88.DBF',
+    'f1_steam': 'F1_89.DBF',
+    'f1_leased': 'F1_90.DBF',
+    'f1_sbsdry_detail': 'F1_91.DBF',
+    'f1_plant': 'F1_92.DBF',
+    'f1_long_term_debt': 'F1_93.DBF',
+    'f1_106_2009': 'F1_106_2009.DBF',
+    'f1_106a_2009': 'F1_106A_2009.DBF',
+    'f1_106b_2009': 'F1_106B_2009.DBF',
+    'f1_208_elc_dep': 'F1_208_ELC_DEP.DBF',
+    'f1_231_trn_stdycst': 'F1_231_TRN_STDYCST.DBF',
+    'f1_324_elc_expns': 'F1_324_ELC_EXPNS.DBF',
+    'f1_325_elc_cust': 'F1_325_ELC_CUST.DBF',
+    'f1_331_transiso': 'F1_331_TRANSISO.DBF',
+    'f1_338_dep_depl': 'F1_338_DEP_DEPL.DBF',
+    'f1_397_isorto_stl': 'F1_397_ISORTO_STL.DBF',
+    'f1_398_ancl_ps': 'F1_398_ANCL_PS.DBF',
+    'f1_399_mth_peak': 'F1_399_MTH_PEAK.DBF',
+    'f1_400_sys_peak': 'F1_400_SYS_PEAK.DBF',
+    'f1_400a_iso_peak': 'F1_400A_ISO_PEAK.DBF',
+    'f1_429_trans_aff': 'F1_429_TRANS_AFF.DBF',
+    'f1_allowances_nox': 'F1_ALLOWANCES_NOX.DBF',
+    'f1_cmpinc_hedge_a': 'F1_CMPINC_HEDGE_A.DBF',
+    'f1_cmpinc_hedge': 'F1_CMPINC_HEDGE.DBF',
+    'f1_email': 'F1_EMAIL.DBF',
+    'f1_rg_trn_srv_rev': 'F1_RG_TRN_SRV_REV.DBF',
+    'f1_s0_checks': 'F1_S0_CHECKS.DBF',
+    'f1_s0_filing_log': 'F1_S0_FILING_LOG.DBF',
+    'f1_security': 'F1_SECURITY.DBF'
+}
+
+DBF_TYPES = {
```

Comment:
If ever DBF is needed for use outside of FERC 1, I would suggest moving a lot of this functionality to `pudl.metadata`. For example, `classes.Package/Resource.from_dbf()` followed by `Package/Resource.to_sql()`. In other words, converting the DBF database to a pudl metadata representation, then exporting out to the desired format.

Reply:
Hmm, that's a nice idea. The FERC Form 2 (like Form 1 but for gas utilities) will also use all of this functionality, so there's definitely some generalization to be done in the near future. It's a little bit specific to FERC's habit of publishing a totally separate database for each year, which we then concatenate together, building a schema out the union of all the database schemas (which for FERC 1 has changed over time but only in an additive way)